scheduler/plugin/wrappedplugin.go

```diff
@@ -0,0 +1,270 @@
+	originalScorePlugin          framework.ScorePlugin
+	filterPluginExtender         FilterPluginExtender
+	scorePluginExtender          ScorePluginExtender
+	normalizeScorePluginExtender NormalizeScorePluginExtender
+	weight                       int32
+	store                        Store
+}
+
+const (
+	pluginSuffix       = "Wrapped"
+	resultBeforePrefix = "Before"
+	resultAfterPrefix  = "After"
+)
+
+func pluginName(pluginName string) string {
+	return pluginName + pluginSuffix
+}
+
+// newWrappedPlugin makes wrappedPlugin from score or/and filter plugin.
+func newWrappedPlugin(s Store, p framework.Plugin, weight int32, opts ...Option) framework.Plugin {
+	options := options{}
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+	plg := &wrappedPlugin{
+		name:   pluginName(p.Name()),
+		weight: weight,
+		store:  s,
+	}
+	if options.extenderOption.filterPluginExtender != nil {
+		plg.filterPluginExtender = options.extenderOption.filterPluginExtender
+	}
+	if options.extenderOption.scorePluginExtender != nil {
+		plg.scorePluginExtender = options.extenderOption.scorePluginExtender
+	}
+	if options.extenderOption.normalizeScorePluginExtender != nil {
+		plg.normalizeScorePluginExtender = options.extenderOption.normalizeScorePluginExtender
+	}
+
+	fp, ok := p.(framework.FilterPlugin)
+	if ok {
+		plg.originalFilterPlugin = fp
+	}
+
+	sp, ok := p.(framework.ScorePlugin)
+	if ok {
+		plg.originalScorePlugin = sp
+	}
+
+	return plg
+}
+
+func (w *wrappedPlugin) Name() string { return w.name }
+func (w *wrappedPlugin) ScoreExtensions() framework.ScoreExtensions {
+	if w.originalScorePlugin != nil && w.originalScorePlugin.ScoreExtensions() != nil {
+		return w
+	}
+	return nil
+}
+
+// NormalizeScore wraps original NormalizeScore plugin of Scheduler Framework.
+// Before and after the execution of original NormalizeScore plugin,
+// we will run arbitrary processing as functions from normalizeScorePluginExtender.
+func (w *wrappedPlugin) NormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	if w.originalScorePlugin == nil || w.originalScorePlugin.ScoreExtensions() == nil {
+		// return nil not to affect scoring
+		return nil
+	}
+
+	if w.normalizeScorePluginExtender != nil {
+		if s := w.doBeforeNormalizeScoreAndStoreResult(ctx, state, pod, scores); !s.IsSuccess() {
+			return s
+		}
+	}
+	orignstatus := w.originalScorePlugin.ScoreExtensions().NormalizeScore(ctx, state, pod, scores)
+	if !orignstatus.IsSuccess() {
+		klog.Errorf("failed to run normalize score: %v, %v", orignstatus.Code(), orignstatus.Message())
+		// If it is not nil, we wiil return the results of AfterNormalizeScore.
+		if w.normalizeScorePluginExtender != nil {
+			return w.doAfterNormalizeScoreAndStoreResult(ctx, state, pod, scores, orignstatus)
+		}
+		// Otherwise, we will return the original results.
+		return orignstatus
+	}
+
+	for _, s := range scores {
```

Comment:
Let's add TODO comment

```suggestion

	// TODO: move to AfterNormalizeScore.
	for _, s := range scores {
```

Reply:
I should move it to AfterNormalizeScore?
This code has been in place here since before I made the change. 
And I think the result of `NormalizeScore` function should be stored before calling `AfterNormalizeScore` , if NormalizeScore is successful

Reply:
> And I think the result of NormalizeScore function should be stored before calling AfterNormalizeScore , if NormalizeScore is successful

Why? 
If we don't move, users need to use `store` even if they don't need to record scores. 
In this PR, it's ok not to move them since the change on this PR will be bigger. But in the future, we should move them out of wrappedPlugin.

Reply:
Ah, you mean that change the current implementation logic?
My above opinion had been based on no change in the current logic. (Currently, users need to use the store if `NormalizeScore` function returns successfully.)